### DISCUSSION TITLE
How to consume head of a string?

### DISCUSSION THREAD

**Discussion by oovm** [rust-lang/regex#958] [Q&A] [answered]
I'm writing a lexer that needs to consume the head of a string based on a pattern.

For example, for `[1-9]+`

| input | match | offset | 
| :- | :- | :- |
| `123abc` |  123  | 3 |
| `0123abc` | × | 

Which method on `Regex` should I use to reject the second input?

**Comment by BurntSushi** [comment, marked as answer]
Change your regex to `^[1-9]+`?

**Reply by oovm** [reply]
Yes, this works, but I can't simply format as`^({})` in preprocessing, some rules may use `(?mode)` or something.

**Reply by BurntSushi** [reply]
I don't see the problem. Can you please provide an example that doesn't work?

**Comment by oovm** [comment]
You are right, I found that what I encountered was due to double escaping, thank you
